File: rust/examples/signer_options/private_key_string.rs
use dotenv::from_filename;
use near_api::prelude::{AccountId, NearToken, NetworkConfig, Signer, Tokens};
use near_crypto::SecretKey;
use std::str::FromStr;

#[tokio::main]
async fn main() {
    from_filename("../.env").unwrap();
    let private_key_string = std::env::var("PRIVATE_KEY").unwrap();
    let account_id_string = std::env::var("ACCOUNT_ID").unwrap();

    // Create a signer from a private key string
    let private_key = SecretKey::from_str(&private_key_string).unwrap(); // ed25519:5Fg2...
    let signer = Signer::new(Signer::secret_key(private_key)).unwrap(); // Create the signer

    let network = NetworkConfig::testnet();

    let account_id: AccountId = account_id_string.parse().unwrap();

    // Test the signer by transferring NEAR
    Tokens::of(account_id.clone()) // example-account.testnet
        .send_to("receiver-account.testnet".parse().unwrap())
        .near(NearToken::from_near(1))
        .with_signer(signer.clone())
        .send_to(&network)
        .await
        .unwrap()
        .assert_success();
}

File: near-kit/examples/keystore-options/private-key-string.js
import { Near } from "near-kit";
import dotenv from "dotenv";

// Load environment variables
dotenv.config();

// Create Near client with private key directly from environment
const near = new Near({
  network: "testnet",
  privateKey: process.env.PRIVATE_KEY, // ed25519:5Fg2...
  defaultSignerId: process.env.ACCOUNT_ID, // example-account.testnet
});

// Test by sending NEAR
const result = await near.send("receiver-account.testnet", "0.1 NEAR");
console.log("Transfer result:", result);
